ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/fabric_ip_addressing.schema.yml

```diff
@@ -25,3 +25,19 @@ keys:
               - "first_id"
               - "odd_id"
               - "same_subnet"
+      vtep:
+        type: dict
+        keys:
+          algorithm:
+            type: str
+            default: "first_id"
+            description: |
+              This variable defines the VTEP IP allocation algorithm used.
+              Each VTEP Source Interface will get a /32 IP address allocated from the relevant vtep_loopback_ipv4_pool via a calculated offset.
+              The offset is calculated using one of the following algorithms:
+                - first_id: `(mlag_primary_id - 1) * 2` where `mlag_primary_id` is the ID of the first node defined under the node_group.
+                  This allocation method will skip every other /31 subnet making it less space efficient than `odd_id`.
+                - odd_id: `(odd_id - 1) / 2`. Requires the node_group to have a node with an odd ID and a node with an even ID.
```

Comment:
The `odd_id` would not work in non-MLAG cases, and even if we ignored it in those cases, it could lead to overlapping IPs if you have:
- MLAG pair A - ID 2, 3 -> offset 1
- Stand-alone Leaf - ID 4 -> offset 4
- MLAG pair B - ID 9, 10 -> offset 4

Reply:
We could enforce that stand-alone leaf switches must use an odd ID, but it would just lead to more waste on regular loopback pools and uplinks.

Reply:
Yes you are definitely right.  I think the idea was that odd_id would not be the default but could be selected by a user so long as they understood that devices in the fabric should be ID'd under the assumption that each switch getting a VTEP IP is in an MLAG pair (even if not all leaf switches are actually in an MLAG pair).

So if you had 3 groups of leafs where group 1 consisted of 2 leafs, group 2 consisted of 1 leaf, and group 3 consisted of 2 leafs, the IDs we would assume the user would use are:

MLAG pair A - ID 1, 2 -> offset 0
Standalone Leaf - ID 3 -> offset 1
MLAG pair B - ID 5, 6 -> offset 2

Reply:
I see what you're saying though that this will lead to more waste on regular loopback and uplink pools